Configurable default gas for eth_call

bmino opened the pull request:
**Description**:

This PR exposes the internal `defaultGas` parameter via an environment variable in order to support overriding the default 400_000 gas to a custom value. This supports consumers who cannot supply a gas amount for eth_call requests and perform low cost queries.

**Related issue(s)**:

Fixes #982 

**Notes for reviewer**:
Currently experimenting with a test for modifying static properties of `EthImpl`. Other untested properties are listed below:
* `ETH_CALL_CACHE_TTL`
* `ETH_BLOCK_NUMBER_CACHE_TTL_MS`
* `ETH_GET_BALANCE_CACHE_TTL_MS`

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

AlfredoG87 reviewed (approved):
LGTM 👍

AlfredoG87 reviewed (changes requested):
Please  just  sign  your  commits  👍 
Otherwise looks good 💯

AlfredoG87 reviewed (approved):
LGTM

AlfredoG87 commented on packages/relay/src/lib/eth.ts:
```diff
@@ -58,7 +58,7 @@ export class EthImpl implements Eth {
   static zeroAddressHex = '0x0000000000000000000000000000000000000000';
   static emptyBloom = "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 ...
-  static defaultGas = EthImpl.numberTo0x(constants.TX_DEFAULT_GAS);
+  static defaultGas = EthImpl.numberTo0x(process.env.TX_DEFAULT_GAS ? Number.parseInt(process.env.TX_DEFAULT_GAS) : 400_000);
```
Like this...

```suggestion
  static defaultGas = EthImpl.numberTo0x(process.env.TX_DEFAULT_GAS ? Number.parseInt(process.env.TX_DEFAULT_GAS) : constants.TX_DEFAULT_GAS);
```

Also, unrelated to this I realized today while debugging this class that since this properties are `static` somehow the `process.env` does not have our custom properties by time it loads, so it will always fallback to the default one. 😅 

One way to fix this could be to declare the property as `private readonly`  instead of `static`


```suggestion
  private readonly defaultGas = EthImpl.numberTo0x(process.env.TX_DEFAULT_GAS ? Number.parseInt(process.env.TX_DEFAULT_GAS) : constants.TX_DEFAULT_GAS);
```

<img width="1292" alt="image" src="https://github.com/hashgraph/hedera-json-rpc-relay/assets/123040664/828949a6-ac27-427c-9bb3-54a732339cad">

<img width="1271" alt="image" src="https://github.com/hashgraph/hedera-json-rpc-relay/assets/123040664/2dd672aa-4fcc-454d-a6e8-f76aa815c497">

bmino commented on packages/relay/src/lib/constants.ts:
```diff
@@ -66,7 +66,6 @@ export default {
     ISTANBUL_TX_DATA_NON_ZERO_COST: 16,
     TX_BASE_COST: 21_000,
     TX_HOLLOW_ACCOUNT_CREATION_GAS: 587_000,
-    TX_DEFAULT_GAS: 400_000,
```
Yup makes sense. I removed it because the "default gas value" wasn't a constant anymore but the "default default gas value" still is 😉

bmino commented:
updated everything and added tests for the default gas override feature. still seeing failures with this test `RPC Server: should execute "eth_getTransactionByHash with missing transaction"` with a timeout error but it is also failing for me on the `main` branch without any changes

AlfredoG87 reviewed (approved):
LGTM 👍

bmino commented:
I fixed the rpc batch2 test but am not sure why the other one about HTS token transfers failed

AlfredoG87 reviewed (approved):
LGTM 💯
